schemas/database/002_event_log.sql

```diff
@@ -0,0 +1,8 @@
+CREATE TABLE logs
+(
+    block_index         BYTEA NOT NULL,
+    transaction_index   BYTEA NOT NULL,
+    log_index           BYTEA NOT NULL,
```

Comment:
For block_number to exceed 2^64 within the next 100 years, a blockchain will need to produce blocks at over 5GHz. We won't see that kind of performance anytime soon.

Reply:
My problem with `INT` and `BIGINT` is that they don't match the definition of `Log` in `ethers_core`:
```
    pub block_number: Option<U64>,
    pub transaction_index: Option<U64>,
    pub log_index: Option<U256>,
```

In postgres, `INT` is i32 (vs u64 in `Log`) and `BIGINT` is i64 (vs u256 in `Log`).

I agree that `BYTEA` might be a little wasteful but since we are keeping raw logs anyway, it seemed more fitting?

Reply:
I'm worried that using a non-integer type will make it hard to do range-based queries which we might want to do in the future for various reasons.

Note that Ethereum itself and the underlying JSON-RPC doesn't actually specify the range, just a "quantity" to be encoded in hex. It is in principle unbounded. https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_getlogs

The types in `ethers-core` are quite arbitrary. For comparison, the (now deprecated) open-ethereum (fka parity) client uses `usize`: https://github.com/openethereum/openethereum/blob/6c2d392d867b058ff867c4373e40850ca3f96969/crates/ethcore/types/src/log_entry.rs#L62

So really everyone just picks a reasonably sized type here (the choice of `U256` in ethers-core actually feels wrong to me. U256 is the i-dont-want-to-think-about-it-default in Ethereum ecosystem, so maybe just laziness).

I gave an argument above for the `block_number`, the  other two `*_index` fields are indices in arrays that are bounded by the block size and gas limit, so very small as well.

An argument could be made that we want to be able to cache all the values that can be expressed in `ethers-core` types. I don't think that is very relevant, given that `ethers-core` itself can not represent all the values that could in theory be returned from the `eth_getLogs` call (which are unbounded integers).